### DISCUSSION TITLE
How to get cliff sensors data using ROS2?

### DISCUSSION THREAD

**Discussion by cloudbuzzer** [iRobotEducation/create3_docs#62] [Questions] [answered]
Where is /cliff_state topic?

**Comment by shamlian** [comment]
Where are you finding documentation about this topic? If the robot experiences a cliff, that is exposed in [/hazard_detection](https://iroboteducation.github.io/create3_docs/api/hazards/). I searched the API documentation and I don't see reference to a "/cliff_state" topic.

**Reply by cloudbuzzer** [reply]
The previous iRobot Create 2 has a topic called /cliff_state, and it is of type Int16.
Is there a way I can get a non Boolean data from the Create 3's cliff sensors?

**Reply by shamlian** [reply]
iRobot Create 2 does not support ROS directly, though third party libraries were written to wrap the OI (to which one are you referring?). At the moment, raw cliff data on Create 3 is not exposed over ROS 2, but that is a feature that could be added in an upcoming update.

**Reply by cloudbuzzer** [reply]
Yeah. Sorry. I was talking about the OI wrappers.
Is there any other way I can get the raw cliff data from the Create 3?

**Reply by shamlian** [reply]
I've created an enhancement ticket with your request [here](https://github.com/iRobotEducation/irobot_create_msgs/issues/7)

**Reply by cloudbuzzer** [reply]
Thanks

**Reply by cloudbuzzer** [reply]
What about direct wheel drive, is it possible?

**Reply by shamlian** [reply]
Create 3 does not expose direct PWM access to the wheels. You can command velocities or positions, and the robot will close the loop.

**Reply by shamlian** [reply]
If you haven't already read this, you might find this helpful: https://github.com/iRobotEducation/irobot_create_msgs

**Reply by justinIRBT** [reply]
To determine which cliff, see the documentation of this message https://github.com/iRobotEducation/irobot_create_msgs/blob/main/msg/HazardDetection.msg.  In the hazard detection vector, each hazard has a type and a header.  The header will have the frame id of the hazard that is triggering, so it will tell you which of the 4 cliffs is associated with the triggered hazard.

**Comment by justinIRBT** [comment, marked as answer]
There is no "/cliff_state" topic on Create 3; to determine which cliff is triggering, see the documentation of the [Hazard Detection message](https://github.com/iRobotEducation/irobot_create_msgs/blob/main/msg/HazardDetection.msg). In the hazard detection vector, each hazard has a type and a header. The header will have the frame id of the hazard that is triggering, so it will tell you which of the 4 cliffs is associated with the triggered hazard.
